src/processor/mod.rs

```diff
@@ -141,6 +204,17 @@ impl<D: EventDatabase> EventProcessor<D> {
             .log_event(&pref, &dig, raw, &sigs)
             .map_err(|_| Error::StorageError)?;
 
+        // If delegated event, check its delegator seal.
+        match ilk {
+            EventData::Dip(dip) => {
+                self.validate_seal(dip.seal, pref, &dig, sn)?;
+            }
+            EventData::Drt(drt) => {
+                self.validate_seal(drt.seal, pref, &dig, sn)?;
```

Comment:
these will cause processing to fail if the DIP/DRT comes before the IXN/ROT, when it should instead escrow the delegated event until the right delegating event comes along. in most cases this will be fine though